applicationset/generators/matrix.go

```diff
@@ -54,18 +56,30 @@ func (m *MatrixGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.App
 			return nil, err
 		}
 		for _, b := range g1 {
-			val, err := utils.CombineStringMaps(a, b)
-			if err != nil {
-				return nil, err
+
+			if appSet.Spec.GoTemplate {
+				tmp := map[string]interface{}{}
+				if err := mergo.Merge(&tmp, a); err != nil {
+					return nil, err
+				}
+				if err := mergo.Merge(&tmp, b); err != nil {
+					return nil, err
+				}
+				res = append(res, tmp)
+			} else {
+				val, err := utils.CombineStringMaps(a, b)
+				if err != nil {
+					return nil, err
```

Comment:
Would you mind wrapping these three errors with `fmt.Errorf` to provide a bit of context? Makes debugging a lot easier.